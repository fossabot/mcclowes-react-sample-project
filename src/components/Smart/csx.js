import styled from "styled-components";

export const SmartWrapper = styled.div``;
