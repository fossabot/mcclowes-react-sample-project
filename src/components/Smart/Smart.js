import React from "react";
import { SmartWrapper } from "./csx";
import { Button, Text } from "../UIKit";

const Smart = props => {
  const { children, active, setActive } = props;

  return (
    <SmartWrapper>
      {children}

      <Button
        doClick={() => {
          setActive(!active);
        }}
        text="Click me"
      />

      <Text.Meta>{active ? "Active" : "Not active"}</Text.Meta>
    </SmartWrapper>
  );
};

export default Smart;
