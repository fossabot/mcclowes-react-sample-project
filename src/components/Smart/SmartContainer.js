import React, { useState } from "react";
import Smart from "./Smart";

const SmartContainer = props => {
  const [clicked, setClicked] = useState(false);

  return <Smart clicked={clicked} setClicked={setClicked} {...props} />;
};

export default SmartContainer;
