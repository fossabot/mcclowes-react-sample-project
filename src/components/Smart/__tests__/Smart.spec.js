import Smart from "../Smart";
import React from "react";
import { shallow } from "enzyme";

describe("Smart", () => {
  it("renders default component", () => {
    const wrapper = shallow(<Smart />);

    expect(wrapper).toMatchSnapshot();
  });
});
