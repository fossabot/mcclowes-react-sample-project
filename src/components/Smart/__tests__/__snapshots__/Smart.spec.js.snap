// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smart renders default component 1`] = `
<styled.div>
  <Button
    doClick={[Function]}
    text="Click me"
  />
  <styled.p>
    Not active
  </styled.p>
</styled.div>
`;
